File: 942-di-string-match/942-di-string-match.java
class Solution {
    public int[] diStringMatch(String S) {
        int[] res = new int[S.length() + 1];
        int l = 0, r = S.length();
        for (int i = 0; i < S.length(); i++)
        {
            res[i] = S.charAt(i) == 'I' ? l++ : r--;
        }
        res[S.length()] = l;
        return res;
    }
}

File: 942-di-string-match/942-di-string-match.cpp
class Solution {
public:
    vector<int> diStringMatch(string S) {
       vector<int> res(S.size() + 1);
        int l = 0, r = S.size();
        for (int i = 0; i < S.size(); i++) {
            res[i] = S[i] == 'I' ? l++ : r--;
        }
        res[S.size()] = l;
        return res; 
        
        
    }
};